File: 백준/Silver/10610. 30/30.java
import java.util.*;

public class Main {
    public static void main(String args[]) {
      Scanner s = new Scanner(System.in);
      String str = s.nextLine();
      int len = str.length();
      int[] arr = new int[len];
      for(int i=0;i<len;i++) arr[i] = str.charAt(i) -'0';
      Arrays.sort(arr);
      int sum=0;
      for(int i=0;i<len;i++) sum+=arr[i];
      if(arr[0] != 0||sum%3 != 0) System.out.println(-1);
      else for(int i=len-1;i>=0;i--) System.out.print(arr[i]);
    }
}

File: 백준/Silver/10610. 30/30.py
str = input()
len = len(str)
sum = 0
arr = [0 for _ in range(len)]
for i in range(len):
    arr[i] = int(str[i]) - int('0')
    sum += arr[i]
arr.sort()
if arr[0] or sum%3 != 0 : print(-1)
else:
    for i in range(len-1,-1,-1): print(arr[i], end='')